File: eCommerce.Mobile/ecommerce/lib/models/categorydetail.dart

import 'dart:core';

class CategoryList
{
    List<CategoryDetail> category=[];
}

class CategoryDetail{
  CategoryDetail();

  String category='';
  int categoryLinkId=0;
  List<int> subCategoryLinkIds=[];
  List<String> subCategories=[];
}

File: eCommerce.Models/ControllerModels/CategoryDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace eCommerce.Models.ControllerModels
{
    [NotMapped]
    public class CategoryDetail
    {
        public string Category { get; set; }
        public int CategoryLinkId { get; set; }
        public List<int> SubCategoryLinkIds { get; set; } = new List<int>();
        public List<string> SubCategories { get; set; } = new List<string>();
    }
}
